### PR TITLE
[1 of 2: DCOS-12561] Page Header Breadcrumbs

### PR DESCRIPTION
**Pull request by jfurrow**
This is a refactor of the `ServicesBreadcrumbs` components. It adds some new features:
* If the user is navigating through groups, only the last two groups will appear in the breadcrumb. All other breadcrumbs get "collapsed" into the ellipsis. Clicking the ellipsis navigates one level up.
* When there is not enough horizontal space to display the service health, it gets removed from the DOM.

I also introduced four new breadcrumb components:
* `Breadcrumb`: this wraps the rest of the Breadcrumb components
* `BreadcrumbCaret`: this will render a caret
* `BreadcrumbSupplementalContent`: this content will not shrink by default, and is meant for icons or other supplemental content
* `BreadcrumbTextContent`: this is the primary content; it will shrink and display an ellipsis when truncated

A follow-up PR will implement these new components in the rest of the app. Until then the breadcrumb styling will incorrect.

This styling is intentional for this PR:

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

**Comment by wavesoft** [comment]
Hey @jfurrow , is something like this expected with this PR?

image

**Comment by jfurrow** [comment]
Sorry @wavesoft the screenshot in the description showed something quite different... that is correct, and https://github.com/dcos/dcos-ui/pull/1935 will fix it to look pretty again. Perhaps we shouldn't split up the PRs at all?

**Comment by wavesoft** [comment]
I was thinking that every PR that targets master should not break functionality. I am not sure how this can be achieved here, possibly by splitting this into a different place, such that each part is not changing anything on master?

But personally, I am also OK if we merge this and the next one in the same day, but I don't know how we can enforce this. Therefore it gets my label 👍

File: src/js/components/PageHeaderBreadcrumbs.js [hunk 1]
```diff
@@ -1,30 +1,58 @@
...
+    const shouldTruncateBreadcrumbs = breadcrumbCount > 3;
 
     const breadcrumbElements = breadcrumbs.reduce((memo, breadcrumb, index) => {
+      if (index === 0) {
+        memo.push(this.getCaret('first-caret'));
+      }
+
+      if (shouldTruncateBreadcrumbs && index === breadcrumbCount - 3) {
+        memo.push(
+          this.getCaret('first-caret-forreal'),
```
Comment (nLight) [inline src/js/components/PageHeaderBreadcrumbs.js]: Let's change the name and not use `forreal` 🙏

Comment (jfurrow) [inline src/js/components/PageHeaderBreadcrumbs.js]: Ughhh thank you for catching this, I did not intend to actually commit this, but here we are... 😳

Comment (jfurrow) [inline src/js/components/PageHeaderBreadcrumbs.js]: Done!

**Pull request merged**